### PR TITLE
Handle no javascript on form

### PR DESCRIPTION
**Pull request by nnat425**
- Add noscript tag
- Set interval at every 2 seconds, to check if all fields pass validation.

### CODE REVIEW

File: app/assets/javascripts/sign_up_async_request.js.erb [hunk 1]
```diff
@@ -10,6 +10,7 @@ $( document ).ready(function() {
 			ajaxRequestArray.pop();
 		}
 
+		$(this).attr('validation-passed',false);
```
Comment (LucasCioffi) [inline app/assets/javascripts/sign_up_async_request.js.erb]: A linter would recommend adding a space after the comma.

File: app/assets/javascripts/sign_up_async_request.js.erb [hunk 2]
```diff
@@ -40,18 +41,9 @@ $( document ).ready(function() {
 					if (result.statusCode == 404) {
 
 						$(".spinner").css("display","none");
-
+						$('#user_email').attr('validation-passed',true)
```
Comment (LucasCioffi) [inline app/assets/javascripts/sign_up_async_request.js.erb]: A linter would recommend adding a space after the comma.

File: app/assets/javascripts/sign_up_async_request.js.erb [hunk 3]
```diff
@@ -95,32 +86,52 @@ function validateEmailDomain(email){
...
-	    $('#check_email_div').css("display","none");
+		$('#check_email_div').css("display","none");
 		throw new InvalidEmailDomainError("Invalid e-mail domain")
 	}
 }
 
 
 window.ClientSideValidations.callbacks.element.fail = function(element, message, callback) {
 	callback();
+	$("#sign_up_button").css("background-color","grey");
```
Comment (LucasCioffi) [inline app/assets/javascripts/sign_up_async_request.js.erb]: A linter would recommend adding a space after the comma.

Even though it's gray, is the button still enabled?

File: app/assets/javascripts/sign_up_async_request.js.erb [hunk 4]
```diff
@@ -95,32 +86,52 @@ function validateEmailDomain(email){
...
 	$(element).css('background', '#EEB4B8');
 	$(element).css('border','0.0625rem #45403A solid');
 	var element_id = $(element).attr('id'); 
 	$($('label[for=' + element_id + ']')[1]).attr('id', element_id + "_error")
+	$(element).attr('validation-passed',false)
```
Comment (LucasCioffi) [inline app/assets/javascripts/sign_up_async_request.js.erb]: A linter would recommend adding a space after the comma.

File: app/assets/javascripts/sign_up_async_request.js.erb [hunk 5]
```diff
@@ -95,32 +86,52 @@ function validateEmailDomain(email){
...
 		$($("[for=user_email]")[0]).text('Enter a valid email address ending in "@schools.nyc.gov".')
 		$($('label[for=' + element_id + ']')[0]).attr('id', element_id + "_error")
 	}
 }
 
+
 window.ClientSideValidations.callbacks.element.pass = function(element, callback) {
 	$(element).css('background', '#E8E4E2');
 	$(element).css('border','1px solid #D0CDC8');
+	$(element).attr('validation-passed',true)
```
Comment (LucasCioffi) [inline app/assets/javascripts/sign_up_async_request.js.erb]: A linter would recommend adding a space after the comma.

File: app/assets/javascripts/sign_up_async_request.js.erb [hunk 6]
```diff
@@ -95,32 +86,52 @@ function validateEmailDomain(email){
...
 }
 
 
+// This internal function enables the
+// the submit buttion if all validations are
+// passing.
+setInterval(function(){ 
+	if($('*[validation-passed="true"]').length == 6){
+		$("#sign_up_button").prop("disabled",false);
+		$("#sign_up_button").css("background-color","#af2228")
```
Comment (LucasCioffi) [inline app/assets/javascripts/sign_up_async_request.js.erb]: A linter would recommend adding a space after the comma, here and on line above too.

File: app/assets/stylesheets/signup.css.scss [hunk 7]
```diff
@@ -25,6 +25,11 @@ input[type="text"], input[type="alt_email"], input[type="email"], input[type="pa
   display:none;
 }
 
-#sign_up_button:focusout{
-	background-color:grey;
+#sign_up_button{
+  background-color:grey;
```
Comment (LucasCioffi) [inline app/assets/stylesheets/signup.css.scss]: Interesting to note that "All gray colors are spelled as gray (not grey).  All modern browsers accept both gray and grey, but early versions of Internet Explorer did not recognize grey."
Source: https://www.w3schools.com/colors/colors_shades.asp

File: app/assets/stylesheets/signup.css.scss [hunk 8]
```diff
@@ -25,6 +25,11 @@ input[type="text"], input[type="alt_email"], input[type="email"], input[type="pa
...
+}
+
+#no-js-message{
+  padding-top:20px;
+  color:red;
```
Comment (LucasCioffi) [inline app/assets/stylesheets/signup.css.scss]: A linter would ask for a space after the colon.

File: app/assets/javascripts/sign_up_async_request.js.erb (hunk 3 above)
Comment (nnat425) [reply app/assets/javascripts/sign_up_async_request.js.erb]: Removed that, we didn't need it.

File: app/assets/javascripts/sign_up_async_request.js.erb (hunk 2 above)
Comment (nnat425) [reply app/assets/javascripts/sign_up_async_request.js.erb]: Removed this line.

File: app/assets/javascripts/sign_up_async_request.js.erb (hunk 1 above)
Comment (nnat425) [reply app/assets/javascripts/sign_up_async_request.js.erb]: Fixed.

File: app/assets/javascripts/sign_up_async_request.js.erb (hunk 4 above)
Comment (nnat425) [reply app/assets/javascripts/sign_up_async_request.js.erb]: Removed this line.

File: app/assets/javascripts/sign_up_async_request.js.erb (hunk 5 above)
Comment (nnat425) [reply app/assets/javascripts/sign_up_async_request.js.erb]: Removed this line.

File: app/assets/javascripts/sign_up_async_request.js.erb (hunk 6 above)
Comment (nnat425) [reply app/assets/javascripts/sign_up_async_request.js.erb]: Removed as well.

File: app/assets/stylesheets/signup.css.scss (hunk 8 above)
Comment (nnat425) [reply app/assets/stylesheets/signup.css.scss]: Fixed.

File: app/views/devise/registrations/new.html.erb [hunk 9]
```diff
@@ -1,4 +1,23 @@
 <html lang="en">
+
+<noscript>
+   <h3 id='no-js-message'>This form requires JavaScript. Your browser may not have JavaScript or it may be disabled. Please try again with a JavaScript-enabled browser or email help@mylibrarynyc.org for assistance.
+   </h3>
+
+   <style>
+   
+   form {
```
Comment (LucasCioffi) [inline app/views/devise/registrations/new.html.erb]: We could use a little more indenting here.

File: app/assets/stylesheets/signup.css.scss [hunk 10]
```diff
@@ -25,6 +25,7 @@ input[type="text"], input[type="alt_email"], input[type="email"], input[type="pa
   display:none;
 }
 
-#sign_up_button:focusout{
-	background-color:grey;
+#no-js-message{
+  padding-top:20px ;
+  color:red ;
```
Comment (LucasCioffi) [inline app/assets/stylesheets/signup.css.scss]: `color:red ;` can be `color: red;`
Same for line 29.

**Pull request merged**